### PR TITLE
clean up temp files in tests using ClenFileV2

### PR DESCRIPTION
**Pull request by DiptoChakrabarty**
Signed-off-by: DiptoChakrabarty <diptochuck123@gmail.com>

This PR cleans up the temp files created in the tests using the CleanFileV2 function present in the testutil package of the project

Works on issue #3060

### CODE REVIEW

File: blockindex/indexer_test.go [hunk 1]
```diff
@@ -313,6 +313,7 @@ func TestIndexer(t *testing.T) {
 	})
 	path := "test-indexer"
 	testPath, err := testutil.PathOfTempFile(path)
+	defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline blockindex/indexer_test.go:316]: and here

File: blockindex/indexbuilder_test.go [hunk 2]
```diff
@@ -159,8 +159,8 @@ func TestIndexBuilder(t *testing.T) {
 	testutil.CleanupPath(t, testPath)
 	testutil.CleanupPath(t, indexPath)
 	defer func() {
-		testutil.CleanupPath(t, testPath)
-		testutil.CleanupPath(t, indexPath)
+		testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline blockindex/indexbuilder_test.go:162]: delete line 159 and 160

File: state/factory/factory_test.go [hunk 3]
```diff
@@ -336,6 +338,7 @@ func testCandidates(sf Factory, t *testing.T) {
 
 func TestState(t *testing.T) {
 	testTriePath, err := testutil.PathOfTempFile(triePath)
+	defer testutil.CleanupPathV2(testTriePath)
```
Comment (CoderZhi) [inline state/factory/factory_test.go:341]: move to line 342

File: state/factory/factory_test.go [hunk 4]
```diff
@@ -94,6 +95,7 @@ func TestSnapshot(t *testing.T) {
 func TestSDBSnapshot(t *testing.T) {
 	require := require.New(t)
 	testStateDBPath, err := testutil.PathOfTempFile(stateDBPath)
+	defer testutil.CleanupPathV2(testStateDBPath)
```
Comment (CoderZhi) [inline state/factory/factory_test.go:98]: move to line 99

File: state/factory/factory_test.go [hunk 5]
```diff
@@ -84,6 +84,7 @@ func TestSnapshot(t *testing.T) {
 	require.NoError(sf.Start(ctx))
 	defer func() {
 		require.NoError(sf.Stop(ctx))
+		defer testutil.CleanupPathV2(testTriePath)
```
Comment (CoderZhi) [inline state/factory/factory_test.go:87]: delete `defer`

File: db/trie/mptrie/merklepatriciatrie_test.go [hunk 6]
```diff
@@ -450,6 +450,7 @@ func TestHistoryTrie(t *testing.T) {
 	AccountKVNamespace := "Account"
 	path := "test-history-trie.bolt"
 	testPath, err := testutil.PathOfTempFile(path)
+	defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline db/trie/mptrie/merklepatriciatrie_test.go:453]: move to line 454

File: db/trie/mptrie/merklepatriciatrie_benchmark_test.go [hunk 7]
```diff
@@ -37,6 +37,7 @@ func benchTrieGet(b *testing.B, async, withDB bool) {
 	}
 	if withDB {
 		testPath, err := testutil.PathOfTempFile(fmt.Sprintf("test-kv-store-%t.bolt", async))
+		defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline db/trie/mptrie/merklepatriciatrie_benchmark_test.go:40]: move to line 41

File: db/sql/sqlite3_test.go [hunk 8]
```diff
@@ -33,6 +34,7 @@ func TestSQLite3StorePutGet(t *testing.T) {
 func TestSQLite3StoreTransaction(t *testing.T) {
 	testSQLite3StoreTransaction := TestStoreTransaction
 	testPath, err := testutil.PathOfTempFile(path)
+	defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline db/sql/sqlite3_test.go:37]: move to line 38

File: db/sql/sqlite3_test.go [hunk 9]
```diff
@@ -21,6 +21,7 @@ const (
 func TestSQLite3StorePutGet(t *testing.T) {
 	testRDSStorePutGet := TestStorePutGet
 	testPath, err := testutil.PathOfTempFile(path)
+	defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline db/sql/sqlite3_test.go:24]: move to line 35

File: db/db_test.go [hunk 10]
```diff
@@ -441,7 +441,7 @@ func TestFilter(t *testing.T) {
 	testPath, err := testutil.PathOfTempFile(path)
 	require.NoError(err)
 	testutil.CleanupPath(t, testPath)
-	defer testutil.CleanupPath(t, testPath)
+	defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline db/db_test.go:444]: delete line 443

File: db/db_test.go [hunk 11]
```diff
@@ -240,7 +240,7 @@ func TestDBBatch(t *testing.T) {
 	testPath, err := testutil.PathOfTempFile(path)
 	require.NoError(t, err)
 	testutil.CleanupPath(t, testPath)
-	defer testutil.CleanupPath(t, testPath)
+	defer testutil.CleanupPathV2(testPath)
```
Comment (CoderZhi) [inline db/db_test.go:243]: delete line 242

File: action/protocol/execution/evm/contract_test.go [hunk 12]
```diff
@@ -31,6 +31,7 @@ func TestCreateContract(t *testing.T) {
 	require := require.New(t)
 	ctrl := gomock.NewController(t)
 	testTriePath, err := testutil.PathOfTempFile("trie")
+	defer testutil.CleanupPathV2(testTriePath)
```
Comment (CoderZhi) [inline action/protocol/execution/evm/contract_test.go:34]: move to line 35

File: action/protocol/execution/protocol_test.go [hunk 13]
```diff
@@ -352,6 +352,7 @@ func (sct *SmartContractTest) prepareBlockchain(
 	cfg.Chain.EnableAsyncIndexWrite = false
 	cfg.Genesis.EnableGravityChainVoting = false
 	testTriePath, err := testutil.PathOfTempFile("trie")
+	defer testutil.CleanupPathV2(testTriePath)
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go:355]: move to line 356

File: action/protocol/execution/protocol_test.go [hunk 14]
```diff
@@ -582,10 +583,13 @@ func TestProtocol_Handle(t *testing.T) {
 		}()
 
 		testTriePath, err := testutil.PathOfTempFile("trie")
+		defer testutil.CleanupPathV2(testTriePath)
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go:586]: move to line 587

File: blockchain/integrity/benchmark_test.go [hunk 15]
```diff
@@ -198,16 +198,19 @@ func injectExecution(nonceMap map[string]uint64, ap actpool.ActPool) error {
 func newChainInDB() (blockchain.Blockchain, actpool.ActPool, error) {
 	cfg := config.Default
 	testTriePath, err := testutil.PathOfTempFile("trie")
+	defer testutil.CleanupPathV2(testTriePath)
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply blockchain/integrity/benchmark_test.go]: done

File: blockchain/integrity/benchmark_test.go [hunk 16]
```diff
@@ -198,16 +198,19 @@ func injectExecution(nonceMap map[string]uint64, ap actpool.ActPool) error {
...
 	if err != nil {
 		return nil, nil, err
 	}
 	defer testutil.CleanupPathV2(testTriePath)
 	testDBPath, err := testutil.PathOfTempFile("db")
+	defer testutil.CleanupPathV2(testDBPath)
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply blockchain/integrity/benchmark_test.go]: done

File: blockindex/indexer_test.go (hunk 1 above)
Comment (DiptoChakrabarty) [reply blockindex/indexer_test.go]: done

File: db/db_test.go [hunk 17]
```diff
@@ -58,7 +58,7 @@ func TestKVStorePutGet(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply db/db_test.go:60]: done

File: db/counting_index_test.go [hunk 18]
```diff
@@ -152,7 +152,7 @@ func TestCountingIndex(t *testing.T) {
 	testPath, err := testutil.PathOfTempFile(path)
 	require.NoError(t, err)
 	testutil.CleanupPath(t, testPath)
-	defer testutil.CleanupPath(t, testPath)
+	defer testutil.CleanupPathV2(testPath)
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply db/counting_index_test.go:154]: done

File: blockindex/indexbuilder_test.go (hunk 2 above)
Comment (DiptoChakrabarty) [reply blockindex/indexbuilder_test.go:160]: done

File: blockindex/bloomfilterindexer_test.go [hunk 19]
```diff
@@ -278,6 +278,7 @@ func TestBloomfilterIndexer(t *testing.T) {
 
 	path := "test-indexer"
 	testPath, err := testutil.PathOfTempFile(path)
+	defer testutil.CleanupPathV2(testPath)
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply blockindex/bloomfilterindexer_test.go]: done

File: action/protocol/execution/protocol_test.go [hunk 20]
```diff
@@ -582,10 +583,13 @@ func TestProtocol_Handle(t *testing.T) {
...
 		require.NoError(err)
 		testDBPath, err := testutil.PathOfTempFile("db")
+		defer testutil.CleanupPathV2(testDBPath)
 		require.NoError(err)
 		testIndexPath, err := testutil.PathOfTempFile("index")
+		defer testutil.CleanupPathV2(testIndexPath)
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply action/protocol/execution/protocol_test.go]: done

File: action/protocol/execution/evm/contract_test.go (hunk 12 above)
Comment (DiptoChakrabarty) [reply action/protocol/execution/evm/contract_test.go]: done

File: blockchain/integrity/benchmark_test.go [hunk 21]
```diff
@@ -198,19 +198,16 @@ func injectExecution(nonceMap map[string]uint64, ap actpool.ActPool) error {
...
 		return nil, nil, err
 	}
 	defer testutil.CleanupPathV2(testTriePath)
 	testDBPath, err := testutil.PathOfTempFile("db")
-	defer testutil.CleanupPathV2(testDBPath)
 	if err != nil {
 		return nil, nil, err
 	}
 	defer testutil.CleanupPathV2(testDBPath)
 	testIndexPath, err := testutil.PathOfTempFile("index")
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply blockchain/integrity/benchmark_test.go:208]: moved all cleanups under one defer

File: blockchain/blockdao/blockdao_test.go [hunk 22]
```diff
@@ -442,6 +441,8 @@ func BenchmarkBlockCache(b *testing.B) {
 		defer func() {
 			require.NoError(b, os.RemoveAll(testPath))
```
Comment (Liuhaai) [inline blockchain/blockdao/blockdao_test.go:442]: L442-L443 is unnecessary due to L444-L445

File: action/protocol/staking/candidate_buckets_indexer_test.go [hunk 23]
```diff
@@ -153,7 +153,7 @@ func TestCandidatesBucketsIndexer_PutGetBuckets(t *testing.T) {
 	testPath, err := testutil.PathOfTempFile("test-bucket")
 	require.NoError(err)
 	defer func() {
```
Comment (Liuhaai) [inline action/protocol/staking/candidate_buckets_indexer_test.go:155]: this can be simplified to 
```
defer  testutil.CleanupPathV2(testPath)
```

**Review by Liuhaai — APPROVED** [review]

File: blockchain/blockdao/blockdao_test.go (hunk 22 above)
Comment (DiptoChakrabarty) [reply blockchain/blockdao/blockdao_test.go]: yes removed

File: action/protocol/staking/candidate_buckets_indexer_test.go (hunk 23 above)
Comment (DiptoChakrabarty) [reply action/protocol/staking/candidate_buckets_indexer_test.go]: done

**Review by CoderZhi — APPROVED** [review]
Thanks for this PR! @DiptoChakrabarty

**Pull request merged**